vscode/src/azure/commands.ts

```diff
@@ -19,19 +19,62 @@ import {
 } from "./workspaceActions";
 import { QuantumUris, compileToBitcode } from "./networkRequests";
 import { getQirForActiveWindow } from "../qirGeneration";
+import { targetSupportQir } from "./providerProperties";
 
 const corsDocsUri = "https://github.com/microsoft/qsharp/wiki/Enabling-CORS";
 
+let currentTreeItem: WorkspaceTreeItem | undefined = undefined;
+
 export function initAzureWorkspaces() {
   const workspaceTreeProvider = new WorkspaceTreeProvider();
-  vscode.window.createTreeView("quantum-workspaces", {
+  const treeView = vscode.window.createTreeView("quantum-workspaces", {
```

Comment:
Didn't catch this in the first go around, but this (and other resources) should probably be added to `context.subscriptions` to be disposed of when the extension unloads.

Same with the return value of `onDidChangeSelection`.